plugins/file-open-save-new/core/src/test/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProviderTest.java

```diff
@@ -0,0 +1,215 @@
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class VFSFileProviderTest extends TestCase {
+
+  public void testIsConnectionRoot() {
+
+    VFSFileProvider testInstance = createTestInstance();
+
+    // TEST - simple negative tests before "pvfs://domain"
+    assertFalse( testInstance.isConnectionRoot( null ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( null ) ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( " " ) ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "pvfs" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "pvfs:" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "pvfs:/" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "pvfs://" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "pvfs:///" ) ) );
+
+    // TEST "pvfs://domain"
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some_Connection" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some-Connection" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection123" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://123someConnection123" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some_Connection/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://some-Connection/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection123/" ) ) );
+
+    assertTrue( testInstance.isConnectionRoot( createTestInstance( "pvfs://123someConnection123/" ) ) );
+
+    // TEST now we have past the root domain
+    assertFalse( testInstance.isConnectionRoot( createTestInstance( "pvfs://someConnection/someFolderA" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot(
+        createTestInstance( "pvfs://someConnection/someFolderA/directory2" ) ) );
+
+    assertFalse( testInstance.isConnectionRoot(
+        createTestInstance( "pvfs://someConnection/someFolderA/directory2/randomFileC.txt" ) ) );
+  }
+
+  public void testHasBuckets() {
+
+    VFSFileProvider testInstance = createTestInstance();
+
+    // TEST non VFSConnectionDetails
+    ConnectionDetails mockConnectionDetails  = mock( ConnectionDetails.class );
+    assertFalse( testInstance.hasBuckets( mockConnectionDetails ) );
+
+    // TEST  does not have buckets
+    VFSConnectionDetails mock_NoBuckets_VFSConnectionDetails  = mock( VFSConnectionDetails.class );
```

Comment:
Aren't these variable names with `_` against current rules? Even if allowed, shouldn't it at least be camelCase: `camel_case_camelCase`?

Reply:
No pentaho checkstyle violation

- ![checkstyle_ide_pr](https://github.com/pentaho/pentaho-kettle/assets/2119761/7f0fc52d-89a4-4503-891f-c50776f024ac)

snake case is acceptable in programming world 

- https://www.freecodecamp.org/news/snake-case-vs-camel-case-vs-pascal-case-vs-kebab-case-whats-the-difference

Reply:
"snake case is acceptable in programming world" --- sure, I don't question that.
However, I don't remember ever seeing it for variable/field names in Pentaho code, hence my ask.
I have seen it for test method names only, as a mixture of camel and snake (and not Pascal and Snake).